example/lib/edit_polygon_example.dart

```diff
@@ -58,51 +68,41 @@ class EditPolygonExampleState extends State<EditPolygonExample> {
       points[annotation.id] = point;
     }
 
-    circleManager.dragEvents(onChanged: (context) async {
-      points[context.annotation.id] = context.annotation.geometry;
+    circleManager.dragEvents(onChanged: (annotation) async {
+      points[annotation.id] = annotation.geometry;
       await _addOrUpdateFillLayer();
     });
 
-    await _addOrUpdateFillLayer();
+    final geoJsonData = {
+      "type": "FeatureCollection",
+      "features": [
+        {
+          "type": "Feature",
+          "id": _featureId,
+          "geometry": {
+            "type": "Polygon",
+            "coordinates": [coordinates],
+          }
+        }
+      ]
+    };
+    await mapboxMap.style
+        .addSource(GeoJsonSource(id: _sourceId, data: jsonEncode(geoJsonData)));
+
+    await mapboxMap.style.addLayer(FillLayer(
+      id: "city-fill-layer",
+      sourceId: _sourceId,
+      fillColor: Colors.pink.value,
+      fillOpacity: 0.3,
+    ));
   }
 
   Future<void> _addOrUpdateFillLayer() async {
```

Comment:
This method needs to be renamed now(and the if condition should not be needed anymore since the source is setup in onStyleLoaded) or maybe it could be inlined since it's a one-liner now?